WeatherForecast/WeatherForecast/WeatherHeaderView.swift

```diff
@@ -0,0 +1,114 @@
+        return currentTemperatureLabel
+    }()
+    
+    private lazy var weatherImage: UIImageView = {
+        let weatherImage = UIImageView()
+        weatherImage.translatesAutoresizingMaskIntoConstraints = false
+        weatherImage.contentMode = .scaleToFill
+        return weatherImage
+    }()
+
+    override init(reuseIdentifier: String?) {
+        super.init(reuseIdentifier: reuseIdentifier)
+        configureContents()
+    }
+    
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    private func configureContents() {
+        contentView.addSubview(weatherImage)
+        contentView.addSubview(locationLabel)
+        contentView.addSubview(minMaxTemperatureLabel)
+        contentView.addSubview(currentTemperatureLabel)
+        
+        NSLayoutConstraint.activate([weatherImage.leadingAnchor.constraint(equalTo: contentView.leadingAnchor, constant: 20),
+                                     weatherImage.topAnchor.constraint(equalTo: contentView.topAnchor, constant: -20),
+                                     weatherImage.widthAnchor.constraint(equalToConstant: 70),
+                                     weatherImage.heightAnchor.constraint(equalToConstant: 70),
+                                     locationLabel.leadingAnchor.constraint(equalTo: weatherImage.trailingAnchor, constant: 10),
+                                     locationLabel.topAnchor.constraint(equalTo: contentView.topAnchor, constant: -20),
+                                     minMaxTemperatureLabel.leadingAnchor.constraint(equalTo: locationLabel.leadingAnchor),
+                                     minMaxTemperatureLabel.topAnchor.constraint(equalTo: locationLabel.bottomAnchor, constant: 10),
+                                     currentTemperatureLabel.leadingAnchor.constraint(equalTo: locationLabel.leadingAnchor),
+                                     currentTemperatureLabel.topAnchor.constraint(equalTo: minMaxTemperatureLabel.bottomAnchor, constant: 10)])
+    }
+    
+    func setUpUI(currentWeather: CurrentWeather?, placemark: [CLPlacemark]?) {
+        if let icon = currentWeather?.weather.first?.icon {
+            let imageURL = String(format: "https://openweathermap.org/img/w/%@.png", icon)
+            weatherImage.downloadImage(from: imageURL)
+        }
+        guard let currentWeather = currentWeather else {
+            return
+        }
+        
+        if let placemark = placemark?.first {
+            locationLabel.text = String(placemark.administrativeArea ?? "")
+            + String(placemark.locality ?? "")
+        }
+        minMaxTemperatureLabel.text = changeToCelcius(minimum: currentWeather.main.tempMin,
+                                                      maximum: currentWeather.main.tempMax)
+        currentTemperatureLabel.text = changeToCelcius(to: currentWeather.main.temp)
+    }
+}
+
+extension UIImageView {
+    func downloadImage(from link: String) {
```

Comment:
```suggestion
    func setImage(from link: String) {
```

요런식의 네이밍은 어떨까요? download는 뭔가 return으로 image를 던져줘야 할 것 같은 느낌이 드는것같아요 👀